### PR TITLE
tests for all core resources

### PR DESCRIPTION
**Pull request by milosferencik**

### CODE REVIEW

File: core/src/main/java/cz/fi/muni/pv217/employee/management/client/VacationRESTClient.java [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
 public interface VacationRESTClient {
 
-    @POST
-    @Path("/snap/create")
-    @Consumes(MediaType.TEXT_PLAIN)
-    @Produces(MediaType.TEXT_PLAIN)
-    boolean hasVacation(@QueryParam("id") Long id, @QueryParam("date") LocalDate date);
+    @GET
+    @Path("/employee/{id}/date")
+    boolean hasEmployeeVacationOnDate(@PathParam("id") long id, @QueryParam("date") String date);
```
Comment (pehala) [inline core/src/main/java/cz/fi/muni/pv217/employee/management/client/VacationRESTClient.java]: I don't think vacation service has this endpoint. Am I correct?   @ZoltanFridrich

File: core/src/main/resources/application.properties [hunk 2]
```diff
@@ -20,4 +20,14 @@ quarkus.oidc.credentials.secret=710078a7-7db6-4268-9cfa-3b39f2630412
 quarkus.oidc.tls.verification=none
 
 # Enable Policy Enforcement
-quarkus.keycloak.policy-enforcer.enable=true
+#quarkus.keycloak.policy-enforcer.enable=true
```
Comment (pehala) [inline core/src/main/resources/application.properties]: Is this needed for something?

File: core/src/test/java/cz/fi/muni/pv217/employee/management/resources/WorkdayResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+                        .post("/create");
+        // Check the response
+        response.then()
+                .statusCode(200)
+                .body(    "date", is("2020-12-28"),
+                        "hours", is(8F),
+                        "employee.id", is(1),
+                        "order.id", is(2));
+        // Delete the created workday because transactions suck in integration tests
+        Long id = Long.valueOf((Integer) response.jsonPath().get("id"));
```
Comment (pehala) [inline core/src/test/java/cz/fi/muni/pv217/employee/management/resources/WorkdayResourceTest.java]: I have found the way how to get rid of this in my other pull request !3, it is not required but if you want you can use that method to clean DB

**Review by pehala — CHANGES_REQUESTED** [review]

File: core/src/test/java/cz/fi/muni/pv217/employee/management/resources/WorkdayResourceTest.java (hunk 3 above)
Comment (milosferencik) [reply core/src/test/java/cz/fi/muni/pv217/employee/management/resources/WorkdayResourceTest.java]: Yeah, but my tests had worked when I noticed your PR . I don't want to spend any more time with that.

File: core/src/main/java/cz/fi/muni/pv217/employee/management/client/VacationRESTClient.java (hunk 1 above)
Comment (milosferencik) [reply core/src/main/java/cz/fi/muni/pv217/employee/management/client/VacationRESTClient.java]: I asked Zoltan to add this endpoint

Comment (pehala) [reply core/src/main/java/cz/fi/muni/pv217/employee/management/client/VacationRESTClient.java]: You could also use `/api/vacation/employee/<id>?from=date&to=date` endpoint which is already implemented

File: salary/src/main/resources/application.properties [hunk 4]
```diff
@@ -1,11 +1,28 @@
...
+quarkus.datasource.username = postgres
+quarkus.datasource.password = pass
+quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/salary
 
 # drop and create the database at startup (use `update` to only update the schema)
 quarkus.hibernate-orm.database.generation = update
 
+# core service
+core-service/mp-rest/url=http://localhost:8080/
+core-service/mp-rest/scope=javax.inject.Singleton
```
Comment (pehala) [inline salary/src/main/resources/application.properties]: Scopes can be declared directly in Code with annotations? If you would like for it to be Singleton please replace annotation instead.

File: salary/src/main/java/cz/fi/muni/pv217/employee/management/service/CoreService.java [hunk 5]
```diff
@@ -4,20 +4,12 @@
...
-import javax.ws.rs.PathParam;
-import javax.ws.rs.Produces;
-import javax.ws.rs.QueryParam;
+import javax.ws.rs.*;
 import javax.ws.rs.core.MediaType;
-import java.time.LocalDate;
 import java.util.List;
 
 @Path("/api")
 @RegisterRestClient(configKey="core-api")
```
Comment (pehala) [inline salary/src/main/java/cz/fi/muni/pv217/employee/management/service/CoreService.java]: Dont you want to also change hits key to core-service?

File: salary/src/main/resources/application.properties (hunk 4 above)
Comment (milosferencik) [reply salary/src/main/resources/application.properties]: Yes I know. XStefans used it like this in example project. I can rewrite it, if you want.

File: salary/src/main/java/cz/fi/muni/pv217/employee/management/service/CoreService.java (hunk 5 above)
Comment (milosferencik) [reply salary/src/main/java/cz/fi/muni/pv217/employee/management/service/CoreService.java]: Yes I will change it also here.

File: salary/src/main/resources/application.properties (hunk 4 above)
Comment (pehala) [reply salary/src/main/resources/application.properties]: If you could, since this is a static variable I would much rather see it defined in code

**Pull request merged**